api/tests/python/services/name_request/test_abstract_name_request.py

```diff
@@ -0,0 +1,44 @@
+"""Tests for AbstractNameRequestMixin."""
+import datetime
+
+import pytest
+from dateutil.tz import gettz
+
+from namex.services.name_request import NameRequestService
+from namex.models import Request as RequestDAO
+
+nr_svc = NameRequestService()
+pacific_tz = gettz("UTC")
+
+
+@pytest.mark.parametrize("input_datetime,expected_date", [
+    (datetime.datetime(2021, 1, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2021, 3, 25)),
+    (datetime.datetime(2021, 2, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2021, 4, 25)),
+    (datetime.datetime(2021, 3, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2021, 5, 23)),
+    (datetime.datetime(2021, 4, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2021, 6, 23)),
+    (datetime.datetime(2021, 5, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2021, 7, 23)),
+    (datetime.datetime(2021, 6, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2021, 8, 23)),
+    (datetime.datetime(2021, 7, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2021, 9, 22)),
+    (datetime.datetime(2021, 8, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2021, 10, 23)),
+    (datetime.datetime(2021, 9, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2021, 11, 23)),
+    (datetime.datetime(2021, 10, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2021, 12, 23)),
+    (datetime.datetime(2021, 11, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2022, 1, 23)),
+    (datetime.datetime(2021, 12, 28, 14, 11, 00, tzinfo=pacific_tz), datetime.date(2022, 2, 22))
```

Comment:
The hour and minute should be set to test edge cases.

The current values will work even if the timezone offset is incorrect (7h or 8h) because 11-7=4 is the same day as 11-8=3. (And vice versa for 11+7=18 and 11+8=19).

Reply:
In other words, these should verify that a start date before the clock change should result in an expiry date with the correct time offset after the clock change.

Reply:
fixed, thanks

Reply:
Added edge cases.